Automattic/wp-calypso
# Subscription Manager: implement virtualized list

Related to [#75535](https://github.com/Automattic/wp-calypso/issues/75535)

## Proposed Changes

This adds a virtualized list to the comment list component.

<img width="1569" alt="CleanShot 2023-04-13 at 17 16 27@2x" src="https://user-images.githubusercontent.com/528287/231805923-f9e5b239-1caf-4731-84a0-9681a7f92683.png">


## Testing Instructions

1. Apply this PR and make sure a valid subkey is set
2. Navigate to [http://calypso.localhost:,](http://calypso.localhost:3000/subscriptions/comments)
3. You should see a list of 10K items
4. Scroll it, it should be snappy!

## Does it really reuse cells?

Yes it does!

https://user-images.githubusercontent.com/528287/231208471-19a8b25e-347e-4da7-bfae-ee2240da20e5.mp4




## Downsides

- React Virtualized containers outputs multiple `<div>`s, so I had to let go of the unordered list approach. Too bad, since this is semantically more correct.
- The component in charge of measuring cells (` <CellMeasurer>`) does not like padding, it simply won't include it. To alliviate this, I had to wrap each row in an outer container. Too bad.


## Pre-merge Checklist
